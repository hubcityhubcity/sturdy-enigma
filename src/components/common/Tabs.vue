<template>
  <div class="tabs">
    <div
      v-for="tab in tabs"
      :key="tab.pathName"
      :class="{ 'tab-selected': $route.name === tab.pathName }"
      class="tab"
    >
      <router-link :to="{ name: tab.pathName }">
        {{ tab.displayName }}
      </router-link>
    </div>
  </div>
</template>

<script>
export default {
  name: `tabs`,
  props: {
    tabs: {
      type: Array,
      default: () => []
    }
  },
  data() {
    return {
      tabIndex: 1
    }
  }
}
</script>
<style scoped>
.tabs {
  display: flex;
  margin-left: 2rem;
}

.tabs .tab {
  cursor: pointer;
  margin-right: 1.5rem;
  font-size: 1rem;
  letter-spacing: -1px;
}

.tabs .tab a {
  color: var(--dim);
  display: block;
  font-weight: 500;
  padding-bottom: 1rem;
}

.tabs .tab a:hover {
  color: var(--link);
}

.tabs .tab.tab-selected {
  border-bottom: 2px solid var(--tertiary);
}

.tabs .tab.tab-selected a {
  color: var(--bright);
}
</style>
