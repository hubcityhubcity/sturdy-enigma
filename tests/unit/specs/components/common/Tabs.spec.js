import { shallowMount } from "@vue/test-utils"
import Tabs from "src/components/common/Tabs"

describe(`Tabs`, () => {
  let wrapper
  it(`has the expected html structure`, async () => {
    wrapper = shallowMount(Tabs, {})
    expect(wrapper.findAll(`.tabs .tab`).length).toBe(0)
    expect(wrapper.element).toMatchSnapshot()
  })

  it(`should show links to other pages`, () => {
    wrapper = shallowMount(Tabs, {
      propsData: {
        tabs: [{ pathName: `r1`, displayName: `one` }],
      },
      mocks: {
        $route: {
          name: `r1`,
        },
      },
      stubs: [`router-link`],
    })
    expect(wrapper.findAll(`.tabs .tab`).length).toBe(1)
    expect(wrapper.element).toMatchSnapshot()
  })
})
