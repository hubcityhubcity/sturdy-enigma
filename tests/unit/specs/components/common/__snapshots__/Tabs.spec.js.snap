// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs has the expected html structure 1`] = `
<div
  class="tabs"
/>
`;

exports[`Tabs should show links to other pages 1`] = `
<div
  class="tabs"
>
  <div
    class="tab tab-selected"
  >
    <router-link-stub
      to="[object Object]"
    >
      
      one
    
    </router-link-stub>
  </div>
</div>
`;
